### PR TITLE
Added flag to enable returning http 500 on collection error

### PR DESCRIPTION
**Pull request by mhowell24**
Hi this PR allows you to enable getting the prom http handler to return a 500 when the exporter is unable to connect to Suricata rather than returning a 200 with no metrics. This follows the Prometheus documentations guidance for reporting collection errors [here](https://prometheus.io/docs/instrumenting/writing_exporters/#failed-scrapes). I have put it behind a flag in the interest of not making it a breaking change.

### CODE REVIEW

**Comment by mhowell24** [comment]
Hi @awelzel hope you don't mind me tagging you. Would you mind having a look at this please? Would be great to hear what you guys think about adding this.

**Comment by mhowell24** [comment]
@majst01 hope you don't mind me tagging you. Would you mind having a look at this please? Would be great to hear what you guys think about adding this

**Comment by awelzel** [comment]
@mhowell24 - sorry for the delay. I'm going to look at it today!

> I have put it behind a flag in the interest of not making it a breaking change.

Seems okay to skip the option and do HTTP 500 unconditionally.

**Comment by mhowell24** [comment]
No worries @awelzel thanks for looking. Will remove the flag today!

File: main.go [hunk 1]
```diff
@@ -792,6 +792,18 @@ func (sc *suricataCollector) Collect(ch chan<- prometheus.Metric) {
 
 	if err := sc.client.EnsureConnection(); err != nil {
 		log.Printf("ERROR: Failed to connect to %v", err)
+
+		if *httpErrOnCollectionErr {
+			log.Printf("ERROR: spoiling scrape because of connection failure")
```
Comment (awelzel) [inline main.go:797]: One the option is gone, this message is redundant with the one above I believe.

```suggestion
```

Comment (mhowell24) [reply main.go]: Good point, removed the additional log

File: main.go [hunk 2]
```diff
@@ -792,6 +792,18 @@ func (sc *suricataCollector) Collect(ch chan<- prometheus.Metric) {
...
+			ch <- prometheus.NewInvalidMetric(prometheus.NewDesc(
```
[the remark this answers is not in the document]
Comment (mhowell24) [reply main.go]: Have moved it into the var block as `FailedCollectionDesc`

**Comment by mhowell24** [comment]
@awelzel have made the changes, let me know what you think

**Comment by awelzel** [comment]
Merged, thanks.

**Pull request merged**